docs/source/data/subscriptions.mdx

```diff
@@ -366,159 +393,154 @@ subscription IncrementingNumber {
+You provide these functions as options to the first argument of `useServer`, like so:
 
-</td>
-</tr>
+```ts{5,12}
+useServer(
+  {
+    schema,
+    // As before, ctx is the graphql-ws Context where connectionParams live.
+    onConnect: async (ctx) => {
+      // Check authentication every time a client connects.
+      if (tokenIsNotValid(ctx.connectionParams)) {
+        // You can return false to close the connection  or throw an explicit error
+        throw new Error('Auth token missing!');
+      }
+    },
+    onDisconnect(ctx, code, reason) {
+      console.log('Disconnected!');
+    },
+  },
+  wsServer,
+);
+```
+For more information and examples of using `onConnect` and `onDisconnect`, see the [`graphql-ws` documentation](https://github.com/enisdenjo/graphql-ws#ws-auth-handling).
 
-</tbody>
-</table>
+### Example: Authentication with the Server and Client
 
-### Example: Authentication with `onConnect`
+>If you plan to use subscriptions with Apollo Client, ensure **[both your client and server subscription protocols are consistent](https://www.apollographql.com/docs/react/data/subscriptions/#choice-of-subscription-protocol)** for the subscription library you are using (this example uses the `graphql-ws` library).
 
-On the client, `SubscriptionsClient` supports adding information to `connectionParams` ([example](https://www.apollographql.com/docs/react/data/subscriptions/#5-authenticate-over-websocket-optional)) that will be sent with the first WebSocket message. On the server, all GraphQL subscriptions are delayed until the connection has been fully authenticated and the `onConnect` callback returns a truthy value.
+On the client side, the `GraphQLWsLink` constructor supports adding information to `connectionParams` ([example](https://www.apollographql.com/docs/react/data/subscriptions/#5-authenticate-over-websocket-optional)). Those `connectionParams` are sent to your server when it connects, allowing you to extract information from the client request by accessing `Context.connectionParams`.
 
-Let's suppose we have our `SubscriptionClient` configured like so:
+Let's suppose we create our subscription client like so:
 
 ```js
-new SubscriptionClient(subscriptionUrl, {
-  // ...other options
+import { GraphQLWsLink } from '@apollo/client/link/subscriptions';
+import { createClient } from 'graphql-ws';
+
+const wsLink = new GraphQLWsLink(createClient({
+  url: 'ws://localhost:4000/subscriptions',
   connectionParams: {
-    authorization: clientToken,
+    authorization: user.authToken,
   },
-})
+}));
 ```
 
-The `connectionParams` argument in the `SubscriptionServer`'s `onConnect` callback contains the information passed by the client and can be used to validate user credentials.
+The `connectionParams` argument, containing the information provided by the client, will be passed to your server allowing you to validate the user's credentials.
```

Comment:
Let's put the "containing the..." in parens instead of commas like:

The `connectionParams` argument (which contains the information provided by the client) is passed to...


"to your server allowing you" -> "to your server, enabling you"